sale_automatic_workflow_stock_location_route/models/sale_order.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2024 Akretion (http://www.akretion.com/)
+# @author: Olivier Nibart <olivier.nibart@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def automatic_set_route_on_sol(self):
+        for sale in self:
+            route_id = sale.workflow_process_id.sale_line_route_id
+            if route_id:
+                sale.order_line.filtered(
+                    lambda sol: sol.qty_delivered_method == "stock_move"
+                ).route_id = route_id
+
+    def _action_confirm(self):
+        self.automatic_set_route_on_sol()
```

Comment:
I think it is arguable to force it at confirmation, potentially erasing a route set manually.
At least it can bring confusion (as the user has no idea the route he could set manually won't be actually used).

An other way would be to set the route during the onchange (or make it a computed editable field) depending on workflow_process_id).

I guess it may be wanted to really erase the route on confirm, but it won't be ok for all use cases.